packages/host/app/components/operator-mode/container.gts

```diff
@@ -590,9 +607,60 @@ export default class OperatorModeContainer extends Component<Signature> {
   };
 
   @action updateSubmode(submode: Submode) {
+    switch (submode) {
+      case Submode.Interact:
+        this.onCodePathChange(null);
+        break;
+      case Submode.Code:
+        this.onCodePathChange(new URL(this.lastCardInRightMostStack.id));
+        break;
+    }
+
     this.operatorModeStateService.updateSubmode(submode);
   }
 
+  @action onCodePathChange(codePath: URL | null) {
+    this.operatorModeStateService.updateCodePath(codePath);
+    this.loadCardForCodeMode.perform(codePath);
+  }
+
+  loadCardForCodeMode = restartableTask(async (codePath: URL | null) => {
+    try {
+      if (!codePath) return;
+
+      const realmsToCheck = [
+        this.cardService.defaultURL.href,
+        baseRealm.url,
+        ...otherRealmURLs,
+      ];
+      const isInAnyRealms = await Promise.all(
+        realmsToCheck.map(async (realmURL) => {
+          const realmPath = new RealmPaths(realmURL);
+          return realmPath.inRealm(codePath);
+        }),
+      );
+      if (!isInAnyRealms.includes(true))
+        throw new Error('URL is not in any realms');
+
+      this.cardForCardMode = await this.cardService.loadModel(codePath);
+      this.isCodePathInvalid = false;
+      this.codePathErrorMessage = null;
```

Comment:
generally you'd want to set these values _before_ the async, not after. if you had an error and typed a new URL, the errors should reset as soon as you start typing not after you try to load the card. I haven't gotten that far in my review, but I think that you need an action that the URL text field calls when the value changes, on of the things this action should do is to reset the error state so that as soon as a user starts typing after getting an error, the error state is cleared.